mono/metadata/class.c

```diff
@@ -427,16 +471,20 @@ mono_type_get_name_recurse (MonoType *type, GString *str, gboolean is_recursed,
 			else
 				g_string_append_c (str, '+');
 		} else if (*klass->name_space) {
-			g_string_append (str, klass->name_space);
+			mono_identifier_escape_type_name_chars (str, klass->name_space);
 			g_string_append_c (str, '.');
 		}
 		if (format == MONO_TYPE_NAME_FORMAT_IL) {
-			char *s = strchr (klass->name, '`');
-			int len = s ? s - klass->name : strlen (klass->name);
+			GString *esc = g_string_new ("");
```

Comment:
I don't think MONO_TYPE_NAME_FORMAT_IL should have names escaped.

This is used in loader.c::find_method and might actually break loading.